vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java

```diff
@@ -158,12 +289,250 @@ public void test_updateContentOrder_forNonExistentId() throws BlockDoesNotExistE
         ContentBlock contentBlock1 = new ContentBlock();
         contentBlock1.setId("notARealId");
         contentBlock1.setContentOrder(Integer.valueOf(1));
-        
+
         List<ContentBlock> contentBlocks = new ArrayList<>();
         contentBlocks.add(contentBlock1);
-        
+
         when(contentBlockRepository.findById("notARealId")).thenReturn(Optional.empty());
         managerToTest.updateContentOrder(contentBlocks);
     }
 
+    @Test
+    public void test_getAllContentBlocks_success() throws BlockDoesNotExistException {
+
+        String slideId = "slide1";
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        ContentBlock firstContentBlock = new ContentBlock();
+        firstContentBlock.setId("contentBlockId1");
+        firstContentBlock.setContentOrder(Integer.valueOf(3));
+
+        ContentBlock secondContentBlock = new ContentBlock();
+        secondContentBlock.setId("contentBlockId2");
+        secondContentBlock.setContentOrder(Integer.valueOf(4));
+
+        List<IContentBlock> contentBlocks = new ArrayList<>();
+        contentBlocks.add(firstContentBlock);
+        contentBlocks.add(secondContentBlock);
+        slide.setContents(contentBlocks);
+        when(slideManager.getSlide(slideId)).thenReturn(slide);
+        when(slideRepo.findById("notARealId")).thenReturn(Optional.empty());
```

Comment:
is this needed?

Reply:
?

Reply:
yes cuz when we rearrange the content order changes and this test , tests that

Reply:
when is `slideRepo.findById("notARealId")` called in this test?

Reply:
removed that, cuz it did seem unnecessary.